mne/io/base.py

```diff
@@ -690,7 +690,10 @@ def set_annotations(self, annotations, emit_warning=True):
             self._annotations = None
         else:
             _ensure_annotation_object(annotations)
-            self._annotations = _xxx(annotations, raw=self, emit_warning=emit_warning)
+            # self._annotations = _xxx(annotations, raw=self, emit_warning=emit_warning)
+            new_annotations = annotations.copy()
+            self._annotations = new_annotations.crop(tmin=self.times[0],
+                                                     tmax=self.times[-1])
```

Comment:
the logic that was to set proper cropping should be set here.

https://github.com/mne-tools/mne-python/blob/62ecf9e482bf84a7f58c5fefff766336068f8bce/mne/io/base.py#L728-L730